src/utils/hive-uri.ts

```diff
@@ -1,4 +1,146 @@
+import { get, isArray } from 'lodash';
+const operationsData = require('./operations.json');
+
+/**
+ * checks if uri entered is valid hive uri
+ * Accepts a string
+ * Returns boolean if uri starts with 'hive://'
+ * */
+
 export const isHiveUri = (uri: string) => {
   let trimUri = uri.trim();
   return trimUri.startsWith('hive://');
 };
+
+// check operation array is valid and is a single operation array
+const _checkOpsArray = (ops: any) => {
+  return ops && isArray(ops) && ops.length === 1 && isArray(ops[0]) && ops[0].length === 2;
+};
+
+const findParentKey = (obj, value, parentKey = null) => {
+  for (let key in obj) {
+    if (obj[key] === value) {
+      return parentKey;
+    } else if (typeof obj[key] === 'object') {
+      const foundKey = findParentKey(obj[key], value, key);
+      if (foundKey) {
+        return foundKey;
+      }
+    }
+  }
+  return null;
+};
+
+// get operation name and signer field from operation object
+const getOperationProps = (opName: string) => {
+  const op = get(operationsData, opName, null);
+  if (op) {
+    const signerField = findParentKey(op, '__signer');
+    return {
+      opName: op.name,
+      opAuthority: op.authority || '',
+      signerField,
+    };
+  } else {
+    return null;
+  }
+};
+
+// validate and format amount field in operation to 3 decimal places
+const _formatAmount = (amount: string) => {
+  const splitAmt = amount.split(' ');
+  if (
+    splitAmt.length === 2 &&
+    parseFloat(splitAmt[0]) &&
+    (splitAmt[1] === 'HIVE' || splitAmt[1] === 'HBD')
+  ) {
+    return `${parseFloat(splitAmt[0]).toFixed(3)} ${splitAmt[1]}`;
+  } else {
+    return null;
+  }
+};
+
+/**
+ * Validates parsed data from hive-uri, checks if operation length is not greater than one and __signer is present in operation
+ * Accepts parsed uri from decode method of hive-uri
+ * Returns an object with error status, keys for showing errors, and operation name parsed from operations.json
+ *
+ * */
+export const validateParsedHiveUri = (parsedUri: any, authoritiesMap: Map<string, boolean>) => {
```

Comment:
I like the overall approach, I think this method should be named differently maybe, considering it not only validating to uri, it is also formatting it... something like `convertToOpsArray` or 'getFormattedOpsArray' ... and it should only return opsArray... and throw error as promise to keep things more clean... thoughts..?